pyvrp/cpp/search/LocalSearch.cpp

```diff
@@ -410,18 +424,43 @@ Solution LocalSearch::exportSolution() const
     std::vector<pyvrp::Route> solRoutes;
     solRoutes.reserve(data.numVehicles());
 
+    std::vector<Trip> trips;
+    std::vector<size_t> visits;
+
     for (auto const &route : routes)
     {
         if (route.empty())
             continue;
 
-        std::vector<size_t> visits;
-        visits.reserve(route.size());
+        trips.clear();
+        trips.reserve(route.numTrips());
```

Comment:
Question: why clear here instead of redefining the variable, i.e., 

```suggestion
        std::vector<Trip> trips;
        trips.reserve(route.numTrips());
```
? Idem for `visits`.

Reply:
Clear does not reset the vector's capacity, but redefining does. By reusing the previous trips vector many (most?) of the reserve() calls don't necessarily do anything because a previous iteration is may have already allocated sufficient memory (which we can reuse, we just need to clear it first).

Reply:
Consider:
```cpp
#include <iostream>
#include <vector>

int main()
{
    std::vector<int> items;
    std::cout << items.capacity() << '\n';

    items.reserve(10);
    std::cout << items.capacity() << '\n';

    items.clear();
    std::cout << items.capacity() << '\n';
}
```
with example output:
```
0
10
10
```